cl/search/api_views.py

```diff
@@ -227,16 +230,23 @@ class SearchV4ViewSet(LoggingMixin, viewsets.ViewSet):
     # but folks will need to log in to get past the thresholds.
     permission_classes = (permissions.AllowAny,)
 
+    document_search_classes = {
+        SEARCH_TYPES.RECAP: DocketDocument,
+        SEARCH_TYPES.DOCKETS: DocketDocument,
+        SEARCH_TYPES.RECAP_DOCUMENT: DocketDocument,
+        SEARCH_TYPES.OPINION: OpinionClusterDocument,
+    }
+
     def list(self, request, *args, **kwargs):
         search_form = SearchForm(request.GET, is_es_form=True)
         if search_form.is_valid():
             cd = search_form.cleaned_data
             search_type = cd["type"]
+            search_query = self.document_search_classes[search_type].search()
```

Comment:
I like this approach, but there's potential overlap with the pattern matching block starting at line `276`. Both use elements from the `document_search_classes` dictionary.

Considering this, should we handle a potential `KeyError` exception here?  Line `293` adds a case `_`clause to the pattern matching, but a simple dictionary like `document_search_classes` won't handle unexpected keys.

Reply:
Yeah, you're right. There was a potential `KeyError` for types that are not supported yet, like `pa` and `oa`. I've refactored the code and centralized the supported types in a dictionary called `supported_search_types`, raising the unsupported error earlier. Let me know what you think.